File: src/types/calendar.ts
export interface CalendarEvent {
  id: string;
  summary: string;
  description: string | null;
  location: string | null;
  start_time: number;
  end_time: number;
  all_day: boolean;
  html_link: string | null;
  is_now: boolean;
  is_soon: boolean;
}

export interface CalendarConfig {
  source_type: "ics_file" | "ics_url" | "none";
  ics_path: string | null;
  ics_url: string | null;
}

File: src-tauri/src/models/calendar.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CalendarEvent {
    pub id: String,
    pub summary: String,
    pub description: Option<String>,
    pub location: Option<String>,
    pub start_time: i64,
    pub end_time: i64,
    pub all_day: bool,
    pub html_link: Option<String>,
    pub is_now: bool,
    pub is_soon: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CalendarConfig {
    pub source_type: String, // "ics_file", "ics_url", "none"
    pub ics_path: Option<String>,
    pub ics_url: Option<String>,
}

impl Default for CalendarConfig {
    fn default() -> Self {
        Self {
            source_type: "none".to_string(),
            ics_path: None,
            ics_url: None,
        }
    }
}
